src/transformers/astro.ts

```diff
@@ -0,0 +1,50 @@
+import { UrlParser, UrlTransformer } from "../types.ts";
+import {
+  getNumericParam,
+  setParamIfDefined,
+  toUrl,
+  toCanonicalUrlString
+} from "../utils.ts";
+export {delegateUrl} from "../utils.ts"
+
+export interface AstroParams {
+  href: string;
+  quality?: string | number;
+}
+
+export const parse: UrlParser<AstroParams> = (url) => {
+  const parsedUrl = toUrl(url);
+  const searchParamHref = parsedUrl.searchParams.get("href")
+  const decodedHref = typeof searchParamHref === "string"
+    ? decodeURIComponent(searchParamHref)
+    : new URL(parsedUrl.pathname, parsedUrl.origin).toString()
+  const encodedHref = encodeURIComponent(toCanonicalUrlString(toUrl(decodedHref)));
+  const width = getNumericParam(parsedUrl, "w") || undefined;
+  const height = getNumericParam(parsedUrl, "h") || undefined;
+  const format = parsedUrl.searchParams.get("f") || undefined;
+  const quality = parsedUrl.searchParams.get("q") || undefined;
+
+  return {
+    width,
+    height,
+    format,
+    base: `/_image?href=${encodedHref}`,
+    params: { quality, href: encodedHref },
+    cdn: "astro",
+  };
+};
+
+export const transform: UrlTransformer = (
+  { url: originalUrl, width, height, format, },
+) => {
+  const parsedUrl = toUrl(originalUrl);
+  const href = toCanonicalUrlString(new URL(parsedUrl.pathname, parsedUrl.origin)).toString()
```

Comment:
No need for the `toString()` on here: `toCanonicalUrlString` returns a string

Reply:
Ahh, left over from the refactor—nice catch